TECHPUBS-3096-1.0: Added prerequsite in set_gigabyte_node_bmc_to_fact… (#941)
Cray-HPE/docs-csm: install/set_gigabyte_node_bmc_to_factory_defaults.md

# Set Gigabyte Node BMC to Factory Defaults

> 

Reviewer: ```suggestion

There are cases when a Gigabyte node BMC must be reset to its factory default settings. This page describes when this reset is appropriate, and how to use management scripts and text files to do the reset.

Set the BMC to the factory default settings in the following cases:

- There are problems using the `ipmitool` command and Redfish does not respond.
- There are problems using the `ipmitool` command and Redfish is running.
- When BIOS or BMC flash procedures fail using Redfish.

```

> ## Prerequisites

Reviewer: ```suggestion
## Procedure
```

**Note**: This section refers to scripts that exist only in the PIT environment. If necessary, copy the LiveCD data from a different machine to get these scripts.

> Use the management scripts and text files to reset Gigabyte BMC to factory default settings. Set the BMC to the factory default settings in the following cases:

Reviewer: I am confused. This preface makes it sound like the list which follows is going to be a list of situations in which the BMC should be reset to its factory default settings.

The first few entries do sound like errors you could encounter, and so they make sense as reasons why you may need to do the factory reset:
```markdown
- There are problems using the `ipmitool` command and Redfish does not respond
- There are problems using the `ipmitool` command and Redfish is running
- When BIOS or BMC flash procedures fails using Redfish
```

But the rest of the list entries sound like a combination of how to do the reset, and prerequisites for doing it:
```markdown
- Run the `do_bmc_factory_default.sh` script
- Run `ipmitool -I lanplus -U admin -P password -H BMC_or_CMC_IP mc reset cold` and flash it again after 5 minutes (300 seconds).
- If booted from the PIT node:
  - the firmware packages are located in the HFP package provided with the Shasta release
  - the required scripts are located in `/var/www/fw/river/sh-svr-scripts`
  - The firmware packages are located in the HFP package provided with the Shasta release
  - The required scripts are located in `/var/www/fw/river/sh-svr-scripts`
```

These latter items belong in a different list.

> Use the management scripts and text files to reset Gigabyte BMC to factory default settings. Set the BMC to the factory default settings in the following cases:

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> - There are problems using the `ipmitool` command and Redfish does not respond

Reviewer: ```suggestion
```

> - There are problems using the `ipmitool` command and Redfish is running

Reviewer: ```suggestion
```

> - When BIOS or BMC flash procedures fails using Redfish

Reviewer: ```suggestion
```

>   - Run the `do_bmc_factory_default.sh` script

Reviewer: ```suggestion
```

>   - Run `ipmitool -I lanplus -U admin -P password -H BMC_or_CMC_IP mc reset cold` and flash it again after 5 minutes (300 seconds).

Reviewer: ```suggestion
  - Run `ipmitool -I lanplus -U admin -P password -H BMC_or_CMC_IP mc reset cold` and flash it again after 5 minutes.
```

>   - Run `ipmitool -I lanplus -U admin -P password -H BMC_or_CMC_IP mc reset cold` and flash it again after 5 minutes (300 seconds).

Reviewer: ```suggestion
```

> - If booted from the PIT node:

Reviewer: ```suggestion
If booted from the PIT node:

```

>   - The firmware packages are located in the HFP package provided with the Shasta release

Reviewer: ```suggestion
- The firmware packages are located in the HPE Cray EX HPC Firmware Pack (HFP) provided with the Shasta release.
```

>   - The required scripts are located in `/var/www/fw/river/sh-svr-scripts`

Reviewer: ```suggestion
- The required scripts are located in the `/var/www/fw/river/sh-svr-scripts` directory.
```

> ## Procedure

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

> ### Apply the BMC Factory Command

Reviewer: ```suggestion
```

> 

Reviewer: ```suggestion
```

1. Create a `node.txt` file and add the target node information as shown:

    Example `node.txt` file with two nodes:

>     ```screen

Reviewer: I've seen people using it, but I don't think "screen" is one of the standard values used for markdown fenced code blocks., At least [not according to this file](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml) (which is what GitHub appears to consider standard). I think for these examples, "text" makes sense.
```suggestion
    ```text
```

    10.254.1.11 x3000c0s9b0 ncn-w002
    10.254.1.21 x3000c0s27b0 uan01
    ```

   Example `node.txt` file with one node:

>     ```screen

Reviewer: ```suggestion
    ```text
```

    10.254.1.11 x3000c0s9b0 ncn-w002
    ```

> 2. Use Redfish to reset the BMC to factory default, the BMC is running 12.84.01 or later version, run:

Reviewer: ```suggestion
1. Use Redfish to reset the BMC to factory default.

    - **Option 1:** If the BMC is running version `12.84.01` or later, then run:
```

>     ```bash

Reviewer: ```suggestion
        ```bash
```

>     ncn-w001# sh do_Redfish_BMC_Factory.sh

Reviewer: ```suggestion
        ncn# sh do_Redfish_BMC_Factory.sh
```

>     ```

Reviewer: ```suggestion
        ```
```

>    - Alternatively, use `ipmitool` to reset the BMC to factory defaults:

Reviewer: ```suggestion
   - **Option 2:** Use `ipmitool` to reset the BMC to factory defaults:
```

>     ```bash

Reviewer: ```suggestion
        ```bash
```

>     ncn-w001# sh do_bmc_factory_default.sh

Reviewer: ```suggestion
        ncn# sh do_bmc_factory_default.sh
```

>     ```

Reviewer: ```suggestion
        ```
```

>    - Alternatively, use the power control script:

Reviewer: ```suggestion
   - **Option 3:** Use the power control script:
```

>     ```bash

Reviewer: ```suggestion
        ```bash
```

>     ncn-w001# sh do_bmc_power_control.sh raw 0x32 0x66

Reviewer: ```suggestion
        ncn# sh do_bmc_power_control.sh raw 0x32 0x66
```

>     ncn-w001# sh do_bmc_power_control.sh raw 0x32 0x66

Reviewer: Will those always be the two arguments that should be used? Or is the user supposed to change them somehow based on the specific environment?

Author: I don't know who is working on Gigabyte firmware updates these days, I'll try Trieu, Tan Kim <tan.trieu@hpe.com>

>     ```

Reviewer: ```suggestion
        ```
```

> 3. Wait five minutes (300 seconds) for the BMC and Redfish to initialize.

Reviewer: ```suggestion
1. Wait five minutes (300 seconds) for the BMC and Redfish to initialize.
```

    ```bash

>     ncn-w001# sleep 300

Reviewer: ```suggestion
    ncn# sleep 300
```

    ```

> 4. Add the default login/password to the BMC.

Reviewer: ```suggestion
1. Add the default login and password to the BMC.
```

    ```bash

>     ncn-w001# ncn-w001# sh do_bmc_root_account.sh

Reviewer: ```suggestion
    ncn# sh do_bmc_root_account.sh
```

    ```

> 5. If BMC is version 12.84.01 or later, skip this step. Otherwise, add the default login/password to Redfish.

Reviewer: ```suggestion
1. Add the default login and password to Redfish.

    **IMPORTANT:** If the BMC is version `12.84.01` or later, then **skip this step**. 
```

    ```bash

>     ncn-w001# sh do_Redfish_credentials.sh

Reviewer: ```suggestion
    ncn# sh do_Redfish_credentials.sh
```

    ```

> 6. Make sure the BMC is not in failover mode. Run the script with the `read` option to check the BMC status:

Reviewer: ```suggestion
1. Make sure the BMC is not in failover mode.

    Run the script with the `read` option to check the BMC status:
```

    ```bash

>     ncn-w001# sh do_bmc_change_mode_to_manual.sh read

Reviewer: ```suggestion
    ncn# sh do_bmc_change_mode_to_manual.sh read
```

    ---------------------------------------------------
    [ BMC: 172.30.48.33 ]
    => Manual mode (O)
    ```

>     If the BMC displays `Failover mode`:

Reviewer: ```suggestion
    The BMC is in failover mode if the previous command includes output similar to the following:
```

>     ```bash

Reviewer: ```suggestion
    ```text
```

    [ BMC: 172.30.48.33 ]
    ==> Failover mode (X) <==
    ```

>     Change the BMC back to manual mode.

Reviewer: ```suggestion
    If the BMC is in failover mode, then change the BMC back to manual mode:
```

    ```bash

>     ncn-w001# sh do_bmc_change_mode_to_manual.sh change

Reviewer: ```suggestion
    ncn# sh do_bmc_change_mode_to_manual.sh change
```

    ```

> 7. If the BMC is in a booted management NCN running v1.4+ or v1.3, reapply the static IP address and clear the DHCP address from HSM/KEA.

Reviewer: ```suggestion
1. If the BMC is in a booted management NCN running Shasta v1.3 or later, then reapply the static IP address and clear the DHCP address from HSM/KEA.
```

>     Determine the MAC address in HSM for the DHCP address for the BMC, then delete it from HSM and restart KEA.

Reviewer: ```suggestion
    Determine the MAC address in HSM for the DHCP address for the BMC, delete it from HSM, and restart KEA.
```

> 8. Reboot or power cycle the target nodes.

Reviewer: ```suggestion
1. Reboot or power cycle the target nodes.
```

# Set Gigabyte Node BMC to Factory Defaults

There are cases when a Gigabyte node BMC must be reset to its factory default settings. This page describes when this reset is appropriate, and
how to use management scripts and text files to do the reset.

Set the BMC to the factory default settings in the following cases:

> - There are problems using the `ipmitool` command and Redfish does not respond.

Author: I looked at the oldest 1.3 version on HPESC: https://internal.support.hpe.com/hpesc/public/docDisplay?docId=a00115422en_us&page=Set_Gigabyte_BMC_To_Factory_Defaults.html

It looks like this should be a general Note about when BIOS or BMC procedures fail using Redfish. Changed to a Note.

- There are problems using the `ipmitool` command and Redfish is running.
- When BIOS or BMC flash procedures fail using Redfish.

## Procedure

**Note**: This section refers to scripts that exist only in the PIT environment. If necessary, copy the LiveCD data from a different machine to get these scripts.

> - Run `ipmitool -I lanplus -U admin -P password -H BMC_or_CMC_IP mc reset cold` and flash it again after 5 minutes (300 seconds).

Author: I changed this a Note: based on the original version of this topic on HPESC:
https://internal.support.hpe.com/hpesc/public/docDisplay?docId=a00115422en_us&page=Set_Gigabyte_BMC_To_Factory_Defaults.html

how to use management scripts and text files to do the reset.

Set the BMC to the factory default settings in the following cases:

- There are problems using the `ipmitool` command and Redfish does not respond.
- There are problems using the `ipmitool` command and Redfish is running.
- When BIOS or BMC flash procedures fail using Redfish.

## Procedure

**This section refers to scripts that exist only in the PIT environment.** If necessary, copy the LiveCD data from a different machine to get these scripts.

**Note**: When BIOS or BMC flash procedures fail using Redfish:

- Run the `do_bmc_factory_default.sh` script
- Run `ipmitool -I lanplus -U admin -P password -H BMC_or_CMC_IP mc reset cold` and flash it again after 5 minutes (300 seconds).

If booted from the PIT node:

- The firmware packages are located in the HPE Cray EX HPC Firmware Pack (HFP) provided with the Shasta release.
- The required scripts are located in `/var/www/fw/river/sh-svr-scripts`

### Apply the BMC factory command

1. Create a `node.txt` file and add the target node information as shown:

    Example `node.txt` file with two nodes:

>     ```screen

Reviewer: I've seen people using it, but I don't think "screen" is one of the standard values used for markdown fenced code blocks., At least [not according to this file](https://github.com/github/linguist/blob/master/lib/linguist/languages.yml) (which is what GitHub appears to consider standard). I think for these examples, "text" makes sense.

```suggestion
    ```text
```

    10.254.1.11 x3000c0s9b0 ncn-w002
    10.254.1.21 x3000c0s27b0 uan01
    ```

   Example `node.txt` file with one node:

>     ```screen

Reviewer: ```suggestion
    ```text
```

    10.254.1.11 x3000c0s9b0 ncn-w002
    ```

2. Use Redfish to reset the BMC to factory default.

    - **Option 1:** If the BMC is running version `12.84.01` or later, then run:

    ```bash
        ncn# sh do_Redfish_BMC_Factory.sh
        ```

>    - Alternatively, use `ipmitool` to reset the BMC to factory defaults:

Reviewer: ```suggestion
   - **Option 2:** Use `ipmitool` to reset the BMC to factory defaults:
```

        ```bash

>     ncn-w001# sh do_bmc_factory_default.sh

Reviewer: ```suggestion
        ncn# sh do_bmc_factory_default.sh
```

        ```

   - **Option 3:** Use the power control script:

        ```bash
        ncn# sh do_bmc_power_control.sh raw 0x32 0x66

>        ```

Reviewer: ```suggestion
        ```
```

>        (raw 0x32 0x66 are Gigabyte/AMI vendor specific IPMI commands to reset to factory defaults.)

Reviewer: ```suggestion
       (`raw 0x32 0x66` are Gigabyte/AMI vendor-specific IPMI commands to reset to factory defaults.)
```

3. Wait five minutes (300 seconds) for the BMC and Redfish to initialize.

    ```bash
    ncn# sleep 300
    ```

> 4. Add the default login/password to the BMC.

Reviewer: ```suggestion
4. Add the default login and password to the BMC.
```

# Set Gigabyte Node BMC to Factory Defaults

There are cases when a Gigabyte node BMC must be reset to its factory default settings. This page describes when this reset is appropriate, and
how to use management scripts and text files to do the reset.

Set the BMC to the factory default settings in the following cases:

- There are problems using the `ipmitool` command and Redfish does not respond.
- There are problems using the `ipmitool` command and Redfish is running.
- When BIOS or BMC flash procedures fail using Redfish.

## Procedure

**This section refers to scripts that exist only in the PIT environment.** If necessary, copy the LiveCD data from a different machine to get these scripts.

**Note**: When BIOS or BMC flash procedures fail using Redfish:

- Run the `do_bmc_factory_default.sh` script

> - Run `ipmitool -I lanplus -U admin -P password -H BMC_or_CMC_IP mc reset cold` and flash it again after 5 minutes (300 seconds).

Reviewer: ```suggestion
- Run `ipmitool -I lanplus -U admin -P password -H BMC_or_CMC_IP mc reset cold` and flash it again after five minutes.
```

If booted from the PIT node:

- The firmware packages are located in the HPE Cray EX HPC Firmware Pack (HFP) provided with the Shasta release.
- The required scripts are located in `/var/www/fw/river/sh-svr-scripts`

### Apply the BMC factory command

1. Create a `node.txt` file and add the target node information as shown:

    Example `node.txt` file with two nodes:

    ```text
    10.254.1.11 x3000c0s9b0 ncn-w002
    10.254.1.21 x3000c0s27b0 uan01
    ```

   Example `node.txt` file with one node:

    ```screen
    10.254.1.11 x3000c0s9b0 ncn-w002
    ```

2. Use Redfish to reset the BMC to factory default.

    - **Option 1:** If the BMC is running version `12.84.01` or later, then run:

        ```bash
        ncn# sh do_Redfish_BMC_Factory.sh
        ```

    - **Option 2:** Use `ipmitool` to reset the BMC to factory defaults:

        ```bash

>         ncn-w001# sh do_bmc_factory_default.sh

Reviewer: Nothing about this procedure indicates that it must be run on ncn-w001 specifically. This is likely a holdover from the old days when many tasks were done from there. I don't see why it would be required in this case.

```suggestion
        ncn# sh do_bmc_factory_default.sh
```

        ```

    - **Option 3:** Use the power control script:

        ```bash
        ncn# sh do_bmc_power_control.sh raw 0x32 0x66
        ```

>         (raw 0x32 0x66 are Gigabyte/AMI vendor specific IPMI commands to reset to factory defaults.)

Reviewer: ```suggestion
        (`raw 0x32 0x66` are Gigabyte/AMI vendor-specific IPMI commands to reset to factory defaults.)
```

### Prerequisites

> **Note**: This section refers to scripts that exist only in the PIT environment.  If necessary, copy the LiveCD data from a different machine to get these scripts.

Reviewer: ```suggestion
**Note**: This section refers to scripts that exist only in the PIT environment. If necessary, copy the LiveCD data from a different machine to get these scripts.
```

#### Apply the BMC Factory Command

1.  Create a `node.txt` file and add the target node information as shown:

    Example `node.txt` file with two nodes:

    ```screen
    10.254.1.11 x3000c0s9b0 ncn-w002
    10.254.1.21 x3000c0s27b0 uan01
    ```

    Example `node.txt` file with one node:

    ```screen
    10.254.1.11 x3000c0s9b0 ncn-w002
    ```

> 2. Use Redfish to reset the BMC to factory default, the BMC is running 12.84.01 or later version, run:

Reviewer 2: Should this say, "if the BMC is running..."?

Reviewer: I have the same question. One way or the other, this should be transformed into a proper sentence with a clear meaning, before we merge.

Author: Fixed
